<launch>
    <arg name="serial_no" default=""/>
    <arg name="camera_name" default="camera_$(arg serial_no)"/>

    <arg name="rgb_max_side" default="1920"/>
    <arg name="rgb_min_side" default="1080"/>
    <arg name="rgb_fps" default="6"/>
    <arg name="depth_max_side" default="1280"/>
    <arg name="depth_min_side" default="720"/>
    <arg name="align_depth_to_colour" default="true"/>
    <arg name="imu_enabled" default="false"/>
    <arg name="infrared_enabled" default="false"/>
    <arg name="device_sync" default="false"/>
    <arg name="odom_frame_id" default="odom_frame"/>

    <arg name="tf_prefix" default="$(arg camera_name)"/>
    <arg name="initial_reset" default="false"/>
    <arg name="fisheye_width" default="$(arg depth_max_side)"/>
    <arg name="fisheye_height" default="$(arg depth_min_side)"/>
    <arg name="enable_fisheye" default="false"/>

    <arg name="depth_width" default="$(arg depth_max_side)"/>
    <arg name="depth_height" default="$(arg depth_min_side)"/>
    <arg name="enable_depth" default="true"/>

    <arg name="infra_width" default="$(arg depth_max_side)"/>
    <arg name="infra_height" default="$(arg depth_min_side)"/>
    <arg name="enable_infra1" default="$(arg infrared_enabled)"/>
    <arg name="enable_infra2" default="$(arg infrared_enabled)"/>

    <arg name="color_width" default="$(arg rgb_max_side)"/>
    <arg name="color_height" default="$(arg rgb_min_side)"/>
    <arg name="enable_color" default="true"/>

    <arg name="fisheye_fps" default="6"/>
    <arg name="depth_fps" default="6"/>
    <arg name="infra_fps" default="6"/>
    <arg name="color_fps" default="$(arg rgb_fps)"/>
    <arg name="gyro_fps" default="400"/>
    <arg name="accel_fps" default="250"/>
    <arg name="enable_gyro" default="$(arg imu_enabled)"/>
    <arg name="enable_accel" default="$(arg imu_enabled)"/>
    <arg name="unite_imu_method" default="none"/>

    <arg name="enable_pointcloud" default="false"/>
    <arg name="enable_sync" default="$(arg device_sync)"/>
    <arg name="align_depth" default="$(arg align_depth_to_colour)"/>

    <arg name="filters" default="disparity,spatial,temporal"/>
    <arg name="json_file_path" default="$(find realsense2_camera)/config/$(arg serial_no).json"/>

    <group ns="$(arg camera_name)">
        <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
            <arg name="odom_frame_id" value="$(arg odom_frame_id)"/>
            <arg name="serial_no" value="$(arg serial_no)"/>
            <arg name="tf_prefix" value="$(arg tf_prefix)"/>
            <arg name="initial_reset" value="$(arg initial_reset)"/>
            <arg name="json_file_path" value="$(arg json_file_path)"/>
            <arg name="enable_pointcloud" value="$(arg enable_pointcloud)"/>
            <arg name="enable_sync" value="$(arg enable_sync)"/>
            <arg name="align_depth" value="$(arg align_depth)"/>

            <arg name="fisheye_width" value="$(arg fisheye_width)"/>
            <arg name="fisheye_height" value="$(arg fisheye_height)"/>
            <arg name="enable_fisheye" value="$(arg enable_fisheye)"/>

            <arg name="depth_width" value="$(arg depth_width)"/>
            <arg name="depth_height" value="$(arg depth_height)"/>
            <arg name="enable_depth" value="$(arg enable_depth)"/>

            <arg name="color_width" value="$(arg color_width)"/>
            <arg name="color_height" value="$(arg color_height)"/>
            <arg name="enable_color" value="$(arg enable_color)"/>

            <arg name="infra_width" value="$(arg infra_width)"/>
            <arg name="infra_height" value="$(arg infra_height)"/>
            <arg name="enable_infra1" value="$(arg enable_infra1)"/>
            <arg name="enable_infra2" value="$(arg enable_infra2)"/>

            <arg name="fisheye_fps" value="$(arg fisheye_fps)"/>
            <arg name="depth_fps" value="$(arg depth_fps)"/>
            <arg name="infra_fps" value="$(arg infra_fps)"/>
            <arg name="color_fps" value="$(arg color_fps)"/>
            <arg name="gyro_fps" value="$(arg gyro_fps)"/>
            <arg name="accel_fps" value="$(arg accel_fps)"/>
            <arg name="enable_gyro" value="$(arg enable_gyro)"/>
            <arg name="enable_accel" value="$(arg enable_accel)"/>
            <arg name="unite_imu_method" value="$(arg unite_imu_method)"/>

            <arg name="filters" value="$(arg filters)"/>
        </include>
    </group>
</launch>
